Amitlatestfaq (#9258)
harness/developer-hub: docs/continuous-delivery/x-platform-cd-features/cd-steps/cd-steps-faqs.md

```

This can happen if the Jenkins Job is parameterized but no parameters are being passed. The reverse, where the Jenkins Job is not parameterized but parameters are being passed, can also cause this issue. To fix this, please make sure that the proper parameters are being passed to Jenkins.

> #### Can we create account level variable using terraform provider?

Reviewer: ```suggestion
#### Can we create account level variables using the terraform provider?
```

> We have below terraform resource for creating variables, this can be used for any scope [account/org/project].
>  
> Please check this once and let us know if this helps:
> https://registry.terraform.io/providers/harness/harness/latest/docs/resources/platform_variables
>  
> If you want to create at account level you do not need to pass org and project in the resource configuration.

Reviewer: ```suggestion
We have [terraform resources for creating variables](https://registry.terraform.io/providers/harness/harness/latest/docs/resources/platform_variables
), this can be used for any scope [account/org/project].
 
If you want to create a variable at the account level, do not pass org or project in the resource configuration.
```

> #### Do we have step level execution field for dashboard?
> Currently we capture step level information in fields for dashboard only for jira and approval step type.

Reviewer: ```suggestion
#### Do we have a step level execution field for dashboards?
Currently we capture step level information in fields for dashboards only for the jira and approval step type.
```

> #### How to use harness variables with java operators to get length of array?

Reviewer: ```suggestion
#### How to use harness variables with java operators to get the length of an array?
```

You can use the size construct of the jexl to get this value. Below is a sample usage:
```
<+size(<+<+pipeline.variables.servicenames>.split(',')>)>
```

#### How to identify which inputSet was used for a pipeline execution?

We have an api which fetches the detail of pipeline execution for a give planExecution Id. So you can get the planExecution ID either via api or from UI and invoke the below api end point:
https://apidocs.harness.io/tag/Pipeline-Execution-Details/#operation/getInputsetYamlV2

#### In case of multi-service deployment, is it possible to index the stage names based on service names they are deploying?

You can have this setting enabled at account level where the stages will be indexed with service name, please try this at your end and let us know if this helps:
"Enable Matrix Labels By Name"

> #### What do you specify as Parent folder if you are creating a file in harness file store using api or terraform?

Reviewer: ```suggestion
#### What do you specify as the Parent folder if you are creating a file in the harness file store using an api or terraform?
```

> For root folder the value of "parent_identifier" attribute is "Root". This is inline with the api call that is made.

Reviewer: ```suggestion
For root folder the value of the "parent_identifier" attribute is "Root". This is inline with the api call that is made.
```

> #### How to export all values from a file store file having entries in key value format for a shell script execution.

Reviewer: ```suggestion
#### How can I export all values from a file store file that has entries in a key value format with a shell script execution.
```

> You can access the file content, write to a local file on delegate and use to source it for environment variable. Below is a sample script:

Reviewer: ```suggestion
You can access the file content, write to a local file on the delegate, and use it the source for an environment variable. Below is a sample script:
```

Reviewer: I tried to rewrite `use to source it for environment variable` to what you see above. let me know if the original meaning still holds

```
cat>>.env<<EOF
<+configFile.getAsString("testenv")>
EOF
export $(xargs < .env)
env
```

> #### If harness end point is a vanity url, what to use as the end point in terraform provider configuration?

Reviewer: ```suggestion
#### If the Harness end point is a vanity url, what should the end point in the terraform provider configuration be?
```

> We can use the vanity url as end point, for example "https://VANITY.harness.io/gateway".

Reviewer: ```suggestion
We can use the vanity url as the end point. For example `https://VANITY.harness.io/gateway`.
```
